backend/Functions/Edna.AssignmentLinks/Edna.AssignmentLinks/AssignmentLinksApi.cs

```diff
@@ -74,14 +78,31 @@ public AssignmentLinksApi(ILogger<AssignmentLinksApi> logger, IMapper mapper)
             [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "assignments/{assignmentId}/links/{linkId}")] HttpRequest req,
             [Table(AssignmentLinksTableName)] IAsyncCollector<AssignmentLinkEntity> linksCollector,
             string assignmentId,
-            string linkId)
+            string linkId,
+            [User] UsersClient usersClient)
         {
             string linkJson = await req.ReadAsStringAsync();
             AssignmentLinkDto linkDto = JsonConvert.DeserializeObject<AssignmentLinkDto>(linkJson);
 
             if (linkId != linkDto.Id)
                 return new BadRequestErrorMessageResult("The provided link content doesn't match the path.");
 
+            if (!req.Headers.TryGetUserEmails(out List<string> userEmails))
+            {
+                _logger.LogError("Could not get user email.");
+                return new BadRequestErrorMessageResult("Could not get user email.");
+            }
+
+            _logger.LogInformation($"Getting user information for '{string.Join(';', userEmails)}'.");
+
+            if (userEmails.Count > 0)
+            {
+                User[] allUsers = await usersClient.GetAllUsers(assignmentId);
+                User user = allUsers.FirstOrDefault(member => userEmails.Any(userEmail => (member.Email ?? String.Empty).Equals(userEmail)));

```

Comment:
Does the following help us get rid of `?? String.Empty`?

```suggestion
                User user = allUsers.FirstOrDefault(member => userEmails.Any(userEmail => userEmail.Equals(member.Email)));

```

Reply:
It was specially added by Vipul in one of the previous PRs as a bugfix

Reply:
Oops. Let's leave it as is in that case :). Thanks for letting me know.